react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.newButt = this.newButt.bind(this);
+    this.state = {
+      btnclick: 0
+    };
+  }
+
+  newButt() {
+    this.setState({ btnclick: this.state.btnclick + 1 });
+  }
+
+  render() {
+    let btnClass = 'normal';
+    if (this.state.btnclick <= 2) {
+      btnClass = 'normal';
+    } else if (this.state.btnclick < 6) {
+      btnClass = 'purple';
+    } else if (this.state.btnclick < 9) {
+      btnClass = 'lightpurple';
+    } else if (this.state.btnclick < 12) {
+      btnClass = 'pink';
+    } else if (this.state.btnclick < 15) {
+      btnClass = 'orange';
+    } else if (this.state.btnclick < 18) {
+      btnClass = 'yellow';
+    } else if (this.state.btnclick < 19) {
+      btnClass = 'normal';
+    } else if (this.state.btnclick > 18) {
+      this.setState({ btnclick: 0 });
```

Comment:
This works out fine in this case, but be careful with trying to set state directly within the render method, it can easily create an infinite loop.